lib/discover.coffee

```diff
@@ -14,20 +14,58 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ###
 
-Promise = require('bluebird')
+_ = require('lodash')
 resin = require('resin-sdk')
+{ enumerateServices, findServices } = require('resin-discoverable-services')
+form = require('resin-cli-form')
+{ spinnerPromise } = require('./utils')
 
-exports.findAvahiDevices = Promise.method ->
-	return [
-		{
-			name: 'resin.local'
-			ip: '192.168.1.17'
-		}
-		{
-			name: 'resin2.local'
-			ip: '192.168.1.11'
-		}
-	]
+# Although we only check for 'resin-ssh', we know, implicitly, that ResinOS
+# devices come with 'rsync' installed that can be used over SSH.
+avahiResinSshTag = 'resin-ssh'
+
+exports.discoverLocalResinOsDevices = (timeout = 4000) ->
+	enumerateServices()
+	.then (availableServices) ->
+		services = []
+
+		for s in availableServices
```

Comment:
Thank you @Page-  - this is much slicker 👍  sent PR here - https://github.com/resin-io-modules/resin-sync/pull/83

By the way, although I think we have discussed it before, can you think of any reasons to keep committing `build/` with every PR? This is the current state of many of our modules on `npm`. I'm thinking of removing this step, add `build/` in .gitignore and simply have a prepublish hook to handle transpiling on-the-fly when `npm publish`'ing.  The main reason is to make life easier both to code contributors and reviewers by reducing noise.